# Test for a /tmp/xx leftover from running the giant bash
# test harness. This can happen in the test harness run is
# interrupted with ^C.
# Without this test, the test harness will start failing
# for no real reason.
# Solaris-specific. Not for upstream.
--- tests/run-all	2016-11-20 11:42:26.312258214 +0100
+++ tests/run-all	2016-11-20 11:41:12.703622687 +0100
@@ -1,4 +1,4 @@
-#! /bin/sh
+#!/bin/bash
 
 : ${TMPDIR:=/tmp}
 export TMPDIR
@@ -26,6 +26,11 @@ ${THIS_SH} ./version
 
 rm -f ${BASH_TSTOUT}
 
+if [ -f /tmp/xx ] ; then
+    echo "someone left a /tmp/xx around. I can't test." 
+    exit 1
+fi
+
 echo Any output from any test, unless otherwise noted, indicates a possible anomaly
 
 for x in run-*
@@ -33,7 +38,7 @@ do
 	case $x in
 	$0|run-minimal|run-gprof)	;;
 	*.orig|*~) ;;
-	*)	echo $x ; sh $x ; rm -f ${BASH_TSTOUT} ;;
+	*)	echo $x ; ${THIS_SH} $x ; rm -f ${BASH_TSTOUT} ;;
 	esac
 done
 
